pgml-dashboard/src/templates/docs.rs

```diff
@@ -78,12 +79,10 @@ impl TocLink {
     ///
     /// * `title` - The title of the link.
     ///
-    pub fn new(title: &str, counter: usize) -> TocLink {
-        let id = format!("header-{}", counter);
-
+    pub fn new(title: &str) -> TocLink {
         TocLink {
             title: title.to_string(),
-            id,
+            id: title.to_case(convert_case::Case::Kebab),
```

Comment:
Could you add a unique id here, maybe:

```rust
id: format!("{}-{}", title, counter),
```

This accomplishes both your goal of making the IDs user-friendly and also ensuring that duplicated headers (which are very common in our docs and blog) don't break the ToC.

Reply:
At the very least if you don't like having document-global unique IDs, maybe have a `HashSet` + counter that checks if the header already exists and if it does, append a counter to it? This will replicate the implementation mkdocs has, where if the headers are unique, they are using kebab case, and if not, it starts counting, e.g. `#non-unique-header`, `#non-unique-header-1`, `#non-unique-header-2`, etc.